cpu/kinetis_common/ldscripts/kinetis-base.ld

```diff
@@ -203,25 +205,25 @@ SECTIONS
     __end__ = .;
     PROVIDE(end = .);
 
+    . = ALIGN(4);
     HEAP_SIZE = ORIGIN(sram_u) + LENGTH(sram_u) - STACK_SIZE - .;
 
     .heap (NOLOAD):
     {
-        . = ALIGN(8);
         _heap_start = .;
-        __heap_start = .;
+        PROVIDE(__heap_start = .); /*__heap_start = .; */
         . = . + HEAP_SIZE;
-        . = ALIGN(8);
         _heap_end = .;
+        PROVIDE(__heap_max = .);
     } > sram_u
 
     /* stack section */
     .stack (NOLOAD):
     {
-        . = ALIGN(8);
```

Comment:
The stack must be 8-byte aligned:

http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.faqs/ka4127.html